### PR TITLE
Rest Client Reactive: Support PathParam from BeanParam

### LINKED ISSUE #20027: @PathParam does not work with REST Reactive Client when mixed with BeanParam
### Describe the bug

Using the @BeanParam in a Reactive REST client does not appear to work.  @BeanParams work fine in regular REST methods.

```
ERROR: HTTP Request to /ping/my-client failed, error id: 3aac885c-4d1c-4426-b61f-85fe9805c5bf-1
java.lang.IllegalArgumentException: Path parameter not provided client
	at org.jboss.resteasy.reactive.common.jaxrs.UriBuilderImpl.replaceParameter(UriBuilderImpl.java:621)
UriBuilderImpl.java:621
	at org.jboss.resteasy.reactive.common.jaxrs.UriBuilderImpl.buildCharSequence(UriBuilderImpl.java:553)
```

Because I've hit a couple odd ones before due to naming I changed my reproducer to use a generic name just to make sure and I get the same error.

Reproducer attached:
[beanparam.zip](https://github.com/quarkusio/quarkus/files/7136490/beanparam.zip)

### Expected behavior

Using the reactive rest client I can reuse @BeanParams to stay consistent with "outer" resource calls or just as a way to consolidate the arguments consistently.

### Actual behavior

An error is thrown saying the Path parameter was not provided.  Tracing through it "looks" to be there up until the invocation of the method.

Line 29 - InvocationHandler
`Object result = invoker.invoke(requestContext.getEndpointInstance(), requestContext.getParameters());`

Then it throws the exception...

### How to Reproduce?

Running the test will throw exception

### Quarkus version or git rev

2.1.4.Final

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
fixes #20027

### CODE REVIEW

File: extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java [hunk 1]
```diff
@@ -486,7 +487,7 @@ A more full example of generated client (with sub-resource) can is at the bottom
                 methodIndex++;
 
                 // finding corresponding jandex method, used by enricher (MicroProfile enricher stores it in a field
-                // to later fill in context with corresponding java.lang.reflect.Method
+                // to later fill in context with corresponding java.lang.reflect.Method)
```
Comment (geoand) [inline extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java:490]: You likely want to revert this :)

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        Client client = RestClientBuilder.newBuilder().baseUri(baseUri).build(Client.class);
+        assertThat(client.getWithBeanParam(new MyBeanParam("123"))).isEqualTo("it works!");
+
+    }
+
+    @Path("/my/{id}/resource")
+    public interface Client {
+        @GET
+        String getWithBeanParam(@BeanParam MyBeanParam beanParam);
+    }
```
Comment (geoand) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java:37]: Should we not also add more methods that have `@PathParam` themselves?

File: extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java:490]: no, it's a closing bracket for the one from one line up

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java (hunk 2 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java:37]: 👍🏻 it's worth checking if they mix well together, will do

File: extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java:490]: Oh right. Good catch :)

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java (hunk 2 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java]: done

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
I believe we should backport this. WDYT?

**Comment by michalszynkiewicz** [comment]
It feels safe. But TBH I'm under impression that passing path params in `@BeanParam` is a niche use case

**Comment by michalszynkiewicz** [comment]
But if you think it's important enough, I don't mind :)

**Comment by geoand** [comment]
Let's just keep it mind

**Comment by michalszynkiewicz** [comment]
I already added the backport label :)

**Pull request merged**